server/src/main/java/org/elasticsearch/search/SearchService.java

```diff
@@ -739,10 +759,10 @@ public void openReaderContext(ShardId shardId, TimeValue keepAlive, ActionListen
         });
     }
 
-    final SearchContext createContext(ReaderContext readerContext,
-                                      ShardSearchRequest request,
-                                      SearchShardTask task,
-                                      boolean includeAggregations) throws IOException {
+    protected SearchContext createContext(ReaderContext readerContext,
+                                          ShardSearchRequest request,
+                                          SearchShardTask task,
+                                          boolean includeAggregations) throws IOException {
```

Comment:
The part that we want to avoid executing when the task is cancelled is below. Creating the search context can be costly on the frozen tier or when security is enabled. So it should be simpler to move all the checks here:
`checkCancelled(task, "create_context")` just before creating the search context.

Reply:
++ fixed in f5b7bd1